reactor-core/src/main/java/reactor/core/scheduler/ParallelScheduler.java

```diff
@@ -134,13 +137,7 @@ ScheduledExecutorService pick() {
         ScheduledExecutorService[] a = executors;
         if (a != SHUTDOWN) {
             // ignoring the race condition here, its already random who gets which executor
-            int idx = roundRobin;
-            if (idx == n) {
-                idx = 0;
-                roundRobin = 0;
-            } else {
-                roundRobin = idx + 1;
-            }
+            int idx = ROUND_ROBIN.getAndUpdate(this, i -> (i < (n-1)) ? i + 1 : 0);
```

Comment:
From the comment above:

> // ignoring the race condition here, its already random who gets which executor

This is a benign data race we don't care about. The change adds about 20x the overhead for getting a scheduler.